### ISSUE TITLE
RFC on draft.txt

### ISSUE THREAD

**Issue by johanmalm** [johanmalm/labwc-0.7.2-release-video#1]
I'm thinking of doing something similar to this:

- https://github.com/labwc/labwc-0.6.0-release-video
- https://www.youtube.com/watch?v=guBnx18EQiA

Have drafted a `story.txt` to get some ideas together before turning it into slides.

Any thoughts, ideas or comments?

@Consolatis @jlindgren90 @ahesford @01micko @tokyo4j @jp7677 @Flrian @heroin-moose

**Comment by Flrian** [comment]
Gave it a quick once over and one thing I'm missing is window rules. I think they are a pretty important feature to a lot of people. Not sure how easy it is to show them in a video but they should be mentioned imho.

**Comment by johanmalm** [comment]
Have assembled a first draft.
It's missing some bits, but I think there is enough there for initials comments.
https://youtu.be/CZiEuZE0reI

If anyone fancies supplying a screencast of IME, pipemenus or regions, those would be gratefully received.

Also, thoughts on how the overall videos feels would be welcome. For example, is there too much text. Would it be better to just focus more on screen-casts?

**Comment by Consolatis** [comment]
Very nicely done so far!

You could almost get the idea that we accidentally made yet another tiling wayland compositor :)

Nitpicky feedback dump:
- I think some of the texts / slides are gone too fast, I had to pause the video multiple times to be able to read them
- drop shadows example is missing
- maybe we should mention the wlroots inherited ability to run nested on top of X11 and wayland, that makes testing labwc way easier (and is in fact how I used it for months before feeling comfortable enough to switch completely to wayland). Especially nested on top of X11 should be tested though, It might have regressed slightly since wlroots 0.17.
- the move to edge / grow to edge example could be put a bit later so we have mouse focused examples at the start
- theme / cursor switching via -tweaks could be interesting
- showcasing wlr-randr or wdisplays for setting scale might be interesting

**Comment by ahesford** [comment]
I agree this looks good, but the text is a bit too long and disappears too quickly. I recommend the following rewording of the "basics" slide, in addition to extending the time a bit:

> Let's start with the basics. A Wayland compositor:
> - Coordinates application input and output with the rest of the system
> - Talks to clients via the Wayland protocol
> - Composites off-screen buffers rendered by applications and draws the image to the display
> - Optionally decorates windows and applies effects during composition
>
> Roughly speaking, a Wayland compositor serves the combined roles of an X11 display server, window manager (e.g., Openbox) and compositor (e.g., picom).

The "what's happened" slide can be shortened a bit:

> What has happened since the version Nov 2022 release video for version 0.6.0?
> - Code base has more than doubled, from 10k lines to 21k in `src/`
> - 866 new commits from 56 unique contributors
> - Ten contributors have submitted at least 10 patches
> - Core team has grown to half a dozen active maintainers
> - Many more contributors evaluate ideas, test patches and more. Thank you to all!
> - We've settled on a 10-week release cycle that includes a 3-week cool-down

**Comment by johanmalm** [comment]
Thanks. Will reduce text a bit. Could even consider taking some of it out and adding it to the webpage instead.
Might shorten some of the screen casts too, to make them more to the point.

**Comment by johanmalm** [comment]
https://youtu.be/J_jYCyxiG5U 

Have simplified a lot
Removed the wall of text.
Trimmed most videos.
Added IME + pipemenu videos.

Duration reduced from 5:47 to 3:21

**Comment by 01micko** [comment]
@johanmalm what sort of background music are you looking for?

**Comment by Consolatis** [comment]
Personally I think the synth vibe in https://youtu.be/guBnx18EQiA worked pretty well so something similar would be my choice, other opinions obviously welcome.

Also, maybe the music can fade out at the end rather than the hard cut in the 0.6.0 release video?

**Comment by 01micko** [comment]
> Also, maybe the music can fade out at the end rather than the hard cut in the 0.6.0 release video?

Pretty easy to do with `ffmpeg`
<details>
<summary>snippet</summary>

```shell
secs=$((300 - 18))  # duration of music clip less fade time - get duration with ffprobe
fsecs=18 # max that my ffmpeg-5.1.3 allows - should be enough
ffmpeg -i clip.mp3 -af "afade=t=out:st=$secs:d=$fsecs" clip_out.mp3 
```
</details>

**Comment by johanmalm** [comment]
I'm open to suggestions.
The license needs to be compatible (e.g. CC BY 4.0).
Last time we used https://filmmusic.io

https://filmmusic.io/song/8085-driving-to-the-night

**Comment by 01micko** [comment]
There are plenty of nice free clips on youtube itself, all genré.

**Comment by Consolatis** [comment]
> https://filmmusic.io/song/8085-driving-to-the-night

That one is 404. found a copy at https://commons.wikimedia.org/wiki/File:Driving_to_the_Night_by_Frank_Schroeter.ogg

**Comment by johanmalm** [comment]
How about this one? https://filmmusic.io/song/12297-city-rush-feat-udio

**Comment by 01micko** [comment]
Just MHO but I _think_  :heart: trumps :+1:  - but not by much. Adding a second or two of silence at each end of either clip would fit in perfectly with the video.

**Comment by johanmalm** [comment]
https://youtu.be/gNIj6VU-IH8

It's out. Thanks for all the help.

**Comment by 01micko** [comment]
Yes, saw, commented and :+1:
